apps/src/craft/code-connection/cc-client.js

```diff
@@ -41,21 +62,31 @@ export default class cc_client {
      *
      */
     async_command(command, callback, key) {
-        $.ajax({
-            type: "GET",
-            url: `${baseUrl}${this.port}/${command}`,
-            success: function (data) {
-                console.log(`Command : ${command} success result : ${data.toString()}`);
-                if (key !== null) {
-                    callback(JSON.parse(data)[key]);
-                } else {
-                    callback();
-                }
-            },
-            error: function (jqxhr, textStatus, error) {
-                console.log(`Command : ${command} fail`, error);
-                callback();
-            }
-        });
+        //before ajax
+        if (window.ipcRenderer !== undefined) {
+             window.ipcRenderer.sendToHost('sendToApp', {
+             'api': 'basic',
+             'url': `${baseUrl}${this.port}/${command}`
+             });
+             this.callbackRef = callback;
```

Comment:
is this assignment guaranteed to happen before `responseFromApp` events start getting sent, or do we need an `if (this.callbackRef !== undefined)` somewhere in the handler?

Reply:
That is my understanding. Sending things over the 'sendToApp' channel takes a message to Code Connection which parses the command url, converts it to a format that Minecraft understands, sends it over there, and then ends with sending a response back to Code Studio with the results of the command (such as that it completed successfully, or in some cases where a response is needed, like checking if a specific block was found, returning the result of the check so that conditionals can be properly evaluated). 

Long story short, 'responseFromApp' shouldn't ever be hit unless we're sending something to the App first.